pkg/dashboard/objects/data.go

```diff
@@ -39,6 +40,16 @@ type StatusInfo struct {
 }
 
 func NewDataLayer(ns []string, ver string, cg HelmConfigGetter) (*DataLayer, error) {
+	if len(ns) == 0 {
```

Comment:
On the level of DataLayer object, there's no harm of having empty version. 
Also, validating the emptiness of namespaces is not the responsibility of DataLayer, since it does no manipulations with it.

The check for `cg` can be there, although I'm not a big fan of checking every piece everywhere. It's hard for me to imagine somebody passes `nil` there. And even if passes - that would be quite deliberate by the caller.

Reply:
Thanks @undera, I will drop the namespace and version checks from the DataLayer. I will leave the `cg` check in, just because it probably won't hurt, but like you said, it would be very strange to receive a `nil` config.